axiom/client.go

```diff
@@ -198,8 +212,12 @@ func (c *Client) call(ctx context.Context, method, path string, body, v interfac
 // newRequest creates an API request. If specified, the value pointed to by
 // body will be included as the request body. If it isn't an io.Reader, it will
 // be included as a JSON encoded request body.
-func (c *Client) newRequest(ctx context.Context, method, endpoint string, body interface{}) (*http.Request, error) {
-	rel, err := url.ParseRequestURI(endpoint)
+func (c *Client) newRequest(ctx context.Context, method, path string, body interface{}) (*http.Request, error) {
+	if IsIngestToken(c.accessToken) && path != "/api/v1/tokens/ingest" {
```

Comment:
I assume we don't call newRequest() when doing actual ingest? Otherwise ingest will just return ErrUnprivilegedToken

Reply:
Ups, good catch. This API path here should actually be the ingest path (not the path for ingest-only tokens). Will add some more tests and fix this.